packages/round-manager/src/features/program/ViewProgramPage.tsx

```diff
@@ -29,6 +33,18 @@ export default function ViewProgram() {
     isSuccess: isRoundsFetched
   } = useListRoundsQuery({ address, signerOrProvider: provider, programId: id })
 
+  let [programExists, setProgramExists] = useState(true)
+  let [hasAccess, setHasAccess] = useState(true)
+
+  useEffect(() => {
+    if (isRoundsFetched) {
+      setProgramExists(!!program)
+
+      program?.operatorWallets.includes(address?.toLowerCase()) ? setHasAccess(true) : setHasAccess(false)
```

Comment:
i wasn't able to get an "Access Denied" to appear for the program page - I always got the 404 even though the program existed.

Steps to reproduce:
1. View a program detail page using Wallet 1
2. Switch to Wallet 2, where Wallet 2 is not one of the operators
3. See the 404 page, instead of just the Access Denied page as expected

Since the `GetPrograms`/`GetRounds` queries includes the address, would the subgraph ever return programs/rounds that _don't_ belong to the given address?

Reply:
actually, i am able to get the Access Denied page when I attempt to look at a round where I am not an operator - I noticed that the `ViewRoundPage` doesn't pass in the `address` into the `GetRounds` query, so it is able to retrieve any round by id

and the `ViewProgramPage` _does_ pass in `address` into the query, so any attempt to retrieve a program not matching the address acts the same as non-existing programs

Reply:
Ah okie got what you are pointing out !
Fixed it up 🙌🏼  thanks for pointing this out